File: JavaScript/largest-series-product.js
const letters = require("./helper/letters");

function product(numArray) {
    return numArray.reduce((accumulator, currentValue) => {
        return accumulator * currentValue;
    }, 1);
}

const largestProduct = (series, span) => {
    if (span > series.length) {
        throw new Error('Span must be smaller than string length');
    }

    if (span < 0) {
        throw new Error('Span must be greater than zero');
    }

    if (series.split("").some(x => (0, letters.isLetter)(x))) {
        throw new Error('Digits input must only contain digits');
    }

    let indices = Array.from({ length: span }, (_, i) => i);
    let maxValue = 0;

    for (let i = 0; i < series.length + 1 - span; i++) {
        let substr = series.slice(i, i + span);
        let result = product(Array.from(substr, Number));

        maxValue = Math.max(maxValue, result);

        indices = indices.map(index => index + 1);
    }

    return maxValue;
};

exports.largestProduct = largestProduct;

console.log((0, exports.largestProduct)("123456789", 3)); // 504
console.log((0, exports.largestProduct)("123456789", 4)); // 3024
console.log((0, exports.largestProduct)("123456789", 5)); // 15120

File: C#/Largest Series Product/LargestSeriesProduct.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class LargestSeriesProduct
{
    private static int Product(int[] numArray)
    {
        return numArray.Aggregate(1, (accumaltor, x) => accumaltor * x);
    }

    public static long GetLargestProduct(string digits, int span)
    {
        if (span > digits.Length)
        {
            throw new ArgumentException("Span must be smaller than string length");
        }

        if (span < 0)
        {
            throw new ArgumentException("Span must be greater than zero");
        }

        if (digits.Any(x => char.IsLetter(x)))
        {
            throw new ArgumentException("Digits input must only contain digits");
        }

        var indices = Enumerable.Range(0, span).ToList();   
        int maxValue = 0;

        for (int i = 0; i < digits.Length + 1 - span; i++)
        {
            string substr = digits.Substring(i, span);
            int result = Product(substr.Select(x => Convert.ToInt32(x.ToString())).ToArray());
            substr.Select(x => Convert.ToInt32(x.ToString())).ToArray();

            maxValue = Math.Max(maxValue, result);

            indices = indices.Select(index => index + 1).ToList();
        }

        return maxValue;
    }
}